tests/phpunit/tests/rest-api/class-wp-rest-block-directory-controller-test.php

```diff
@@ -0,0 +1,194 @@
+<?php
+/**
+ * Test WP_REST_Block_Directory_Controller_Test()
+ *
+ * @package Gutenberg
+ * phpcs:disable
+ */
+class WP_REST_Block_Directory_Controller_Test extends WP_UnitTestCase {
+	protected $controller = null;
+
+	function setUp() {
+		parent::setUp();
+
+		global $wp_rest_server;
+		$wp_rest_server = new \WP_REST_Server;
+		$this->server   = $wp_rest_server;
+		do_action( 'rest_api_init' );
+
+		$this->controller = new WP_REST_Block_Directory_Controller();
+		$this->controller->register_routes();
+
+	}
+
+	public function test_register_routes() {
+		$routes = $this->server->get_routes();
+
+		$this->assertArrayHasKey( '/wp/v2/block-directory/search', $routes );
+		$this->assertArrayHasKey( '/wp/v2/block-directory/install', $routes );
+		$this->assertArrayHasKey( '/wp/v2/block-directory/uninstall', $routes );
+	}
+
+	/**
+	 * Tests that an error is returned if the block plugin slug is not provided
+	 */
+	function test_should_throw_no_slug_error() {
+		$request = new WP_REST_Request( 'POST', '/wp/v2/block-directory/install', [] );
+
+		$result = $this->controller->install_block( $request );
```

Comment:
Instead of calling the controller directly, we typically call `rest_do_request()` That way all the server processing occurs like parameter validation.